### ISSUE TITLE
Adaptive HLS

### ISSUE THREAD

**Issue by anxdpanic** [anxdpanic/plugin.video.twitch#283]

**Comment by ghost** [comment]
this might be related to multithreaded segments download.. as a feature to also help against buffering

**Comment by anxdpanic** [comment]
Both of these will rely on Kodi, or a binary add-on in the future to provide that functionality. With the progression with dash/mpd support it does look promising, just not available for hls yet.

**Comment by anxdpanic** [comment]
inputstream.adaptive support added in #326 requires latest script.module.python.twitch master
https://github.com/MrSprigster/Twitch-on-Kodi/releases/tag/2.1.0-alpha

**Comment by greystashh** [comment]
Is it possible to integrate the Time Shift function similar to the YouTube Addon?

**Comment by anxdpanic** [comment]
I don't believe time shift will be possible

**Comment by greystashh** [comment]
Have you tried to?

**Comment by anxdpanic** [comment]
Looking at this more currently, and trying to capture log of a few issues as well. Offending streams and my availability have been what's slowed that progess. Hoping to catch them tonight.

**Comment by greystashh** [comment]
Have you been successful?

**Comment by anxdpanic** [comment]
Have provided peak3d with logs for streams with issues(don't play(non-win)/crash(win)).

Still not looking like time shift will be available, but will keep an eye on it. After we bump to release with adaptive working, I will close this issue and open a new timeshift one.

**Comment by greystashh** [comment]
Do you think that timeshift is not possible because of inputstream hls or is it twitch related issue?

**Comment by anxdpanic** [comment]
At the core it's twitch, there is no way to request a specific segment in live content that I can see. Likely due to the way they shift live content to their separate vod system. May be some(within played time/cache) support for it later when appending segments is possible in inputstream, though I believe that is still quite a ways out.

**Comment by greystashh** [comment]
Has there been any news so far?

**Comment by greystashh** [comment]
Have you already been able to fix some issues?

**Comment by anxdpanic** [comment]
Been no movement, I don't expect timeshift to be anytime soon if at all. It's something I would like to see as well, will be keeping an eye on it. I have noticed that some streams that previously didn't load with adaptive were working which may be due to a few different things.

**Comment by greystashh** [comment]
Do you know the reason why some streams still dont work with adaptive?

**Comment by anxdpanic** [comment]
No, I collected a bunch of the stream metadata(codecs, node, cluster etc.) and there are working examples that match not working examples, and theres no way to tell if player stopped due to error to failover automatically. 
So setting a default quality for the not working streams is the only way around that.

**Comment by greystashh** [comment]
Might it be possible to deactivate adaptive for Vods as a temporary workaround to avoid activate and deactivate inputstream manually?

**Comment by anxdpanic** [comment]
Yea, I also may change the behaviour and make adaptive available to qualities still but only used if it's set as default manually. Seems too often they are not working, and I'd like to get this version to official so I can start work on migrating to helix api.

**Comment by anxdpanic** [comment]
#351 Changes it so adaptive is no longer default if enabled.

**Comment by anxdpanic** [comment]
Updated https://github.com/MrSprigster/Twitch-on-Kodi/releases/tag/2.1.0-alpha with the change, will test for a couple days then go to official.

**Comment by greystashh** [comment]
Could you cancel the change again? With new inputstream versions I havent experienced much errors anymore. Besides it is not very comfortable to select it manually. In general playback without adaptive is laggy. If a video wouldnt play with adaptive you can easily turn inputstream off. I think that is the better solution than now

**Comment by anxdpanic** [comment]
I didn't revert the change, but you can now choose Adaptive as a default quality.
https://github.com/MrSprigster/Twitch-on-Kodi/releases/tag/2.2.1-test

**Comment by anxdpanic** [comment]
@greystashh can you confirm this working for you as well. Next update is expected to be a larger python 3 compat. update, so I will bump this minor if this resolves it. Thank you 👍

**Comment by greystashh** [comment]
Yes, it works now

It would be nice if you can bump it to official repo
